spec/rspec/core_spec.rb

```diff
@@ -129,6 +129,49 @@
     end
   end
 
+  describe ".current_scope" do
+    # puts "i am in the describe block: #{currently_executing_a_context_hook?}"
+
+    unless RSpec.current_scope.nil?
+      raise "bad current scope: #{RSpec.current_scope.inspect}"
+    end
+
+    before :all do
+      # expect(RSpec.current_scope).to eq(:before_all_hook)
+      expect(RSpec.current_scope).to eq(:example_group)
+      # puts "i am in the :all hook: #{currently_executing_a_context_hook?}"
+    end
+
+    before :each do
+      # expect(RSpec.current_scope).to eq(:before_each_hook)
+      expect(RSpec.current_scope).to eq(:example)
+      # puts "i am in the :each hook: #{currently_executing_a_context_hook?}"
+    end
+
+    after :each do
+      # expect(RSpec.current_scope).to eq(:after_each_hook)
+      expect(RSpec.current_scope).to eq(:example)
+      # puts "i am in the :aeach hook: #{currently_executing_a_context_hook?}"
+    end
+
+    after :all do
+      # expect(RSpec.current_scope).to eq(:after_all_hook)
+      expect(RSpec.current_scope).to eq(:example_group)
+      # puts "i am in the :aall hook: #{currently_executing_a_context_hook?}"
+    end
+
+    it "returns :example" do
+      # puts "i am in the spec: #{currently_executing_a_context_hook?}"
+      # puts "this is the actual spec itself"
+      # byebug
+      expect(RSpec.current_scope).to eq(:example)
+    end
+
+    it "works for more than one example in a describe block" do
+      expect(RSpec.current_scope).to eq(:example)
```

Comment:
Wondering if all the above expectations can be made inside a synthetic example group, i.e.:
```ruby
it "..." do
  group = RSpec.describe
  group.before(:context) { expect(RSpec.current_scope).to be(:example_group) }
  # etc
end
```

Reply:
This is possible, but I'm not getting very good feedback when my expectations fail. I'd like to see exactly where it is not yet implemented correctly when it fails.

Synthetic example groups are giving me unrelated errors (from `it_behaves_like 'library wide checks', 'rspec-core'`, i think?).

Also, I found specs for the old implementation. It has some oversights:
```rb
it "does not set currently_executing_a_context_hook? outside of before(:context) execution" do
  group = RSpec.describe
  hook_result = nil

  group.before(:context) { hook_result = group.currently_executing_a_context_hook? }
  # Notice how the method is called on the group assigned to a variable.
  # If the example was realistic it would be called on the value of self and the hook would raise.
  group.before(:each) { hook_result = group.currently_executing_a_context_hook? }
  # We also forgot to test what the method does inside an example... Where it also raises
  group.example("") {}
  group.run
  expect(hook_result).to be false
end
```

So I think it'd be better to dogfood/integration test it.

Would these specs be more appropriately located in the spec/integration folder?
Should I write a .feature too? (which would also help document this things existence)

Reply:
I wrote a cucumber spec for it. Turned out to be the best fit since messing with global configuration caused unrelated failures in the rspec tests.

I still did not find room to use synthetic example groups, since they need to look like this:

```rb
it "bla" do
  yays = []
  group = RSpec.describe
  group.before(:context) do
    expect(RSpec.current_scope).to be(:before_context_hook)
    yays << :bc
  end
  group.after(:context) do
    expect(RSpec.current_scope).to eq(:after_context_hook)
    yays << :ac
  end

  group.it "be like this" do
    expect(RSpec.current_scope).to eq(:example)
    yays << :ex
  end

  group.run

  expect(yays).to eq([:bc, :ex, :ac])
end
```

Which is a bit more cumbersome than I'd like when writing/reading/fixing specs.